docs/source/manual/jooq.rst

```diff
@@ -0,0 +1,120 @@
+      </dependency>
+      ...
+  </dependencies>
+
+Usage
+-----
+
+Install the ThreadLocalJooqModule_ Guice module to enable injecting a DSLContextProvider_ instance that can be used to retrieve a jOOQ DSLContext_ for making database calls. The ThreadLocalJooqModule_ allows accessing a DSLContext_ that assumes database access happens within a single thread; this is the default request lifecycle for JAX-RS requests so this is a safe default. Usage across multiple threads is not supported at this time.
+
+Once the Guice module is installed, any methods that will make calls to the database using the DSLContext_ must be annotated with the JooqTransactional_ annotation. This annotation will start a unit of work and wrap the method call in a transaction. Nested transactions can be done by nesting method calls annotated with JooqTransactional_.
+
+A DataSource_ and jOOQ SQLDialect_ must be bound with Guice for the ThreadLocalJooqModule_ to use.
+
+.. note:: The DSLContextProvider_ must be injected via Guice (rather than manually constructed) for the JooqTransactional_ annotation to work.
+
+Making a database call:
+
+.. code-block:: java
+
+  public class CreateMyModelCommand {
+    private final DSLContextProvider dslContextProvider;
+
+    @Inject
+    CreateMyModelCommand(DSLContextProvider dslContextProvider) {
+      this.dslContextProvider = dslContextProvider;
+    }
+
+    /**
+     * Inserts a new {@link MyModel} into the database.
+     */
+    @JooqTransactional
+    public MyModelRecord execute(String name, String description) {
+      return dslContextProvider.get()
+          .insertInto(MY_MODEL)
+          .set(MY_MODEL.NAME, name)
+          .set(MY_MODEL.DESCRIPTION, description)
+          .returning()
+          .fetchOne();
+    }
+  }
+
+Configuring the required bindings:
+
+.. code-block:: java
+
+  public class JooqConfigurationModule extends AbstractModule {
+  @Override
+    protected void configure() {
+      install(new ThreadLocalJooqModule());
+    }
+
+    /**
+     * Load data source and connection pool information from configuration
+     */
+    @Provides
+    @Singleton
+    DataSource provideDataSource(ImmutableHierarchicalConfiguration configuration) {
+      try {
+        ComboPooledDataSource dataSource = new ComboPooledDataSource();
+        dataSource.setDriverClass(configuration.getString("config.jooq.driver_class"));
+
+        ImmutableHierarchicalConfiguration connectionConfig =
+            configuration.immutableConfigurationAt("config.jooq.connection");
+        dataSource.setJdbcUrl(connectionConfig.getString("url"));
+        dataSource.setUser(connectionConfig.getString("username"));
+        dataSource.setPassword(connectionConfig.getString("password"));
+
+        ImmutableHierarchicalConfiguration c3p0Config =
+            configuration.immutableConfigurationAt("config.jooq.c3p0");
+        dataSource.setMinPoolSize(c3p0Config.getInt("min_size"));
```

Comment:
The configuration that is being read is not documented anywhere. Also, I would have a default value if one does not exist in the config.

Reply:
This is just documentation of an example of configuring a datasource. Other datasources and/or pools can be used besides C3P0, and can be configured however services prefer.

I updated to clarify that this is just an example: https://github.com/cerner/beadledom/commit/4861647139709047ffe28615902ce61ba8a28013#diff-2a3927f61725d9baa48fb9c016e79882L66